src/server.py

```diff
@@ -34,22 +38,23 @@ def server():
                 message_complete = True
 
         try:
-            parse_request(recv_message.decode('utf-8'))
+            client_uri = parse_request(recv_message.decode('utf-8'))
         except TypeError as msg:
             reply = response_error(405, str(msg))
         except ValueError as msg:
             reply = response_error(400, str(msg))
         else:
-            reply = response_ok()
-
+            resolved = resolve_uri(client_uri)
+            reply = response_ok(resolved)
```

Comment:
according to my reading below, the value of `resolved` returned from `resolve_uri` above *might* be a tuple containing content and mime-type, or it *might* be a full on error response.  I don't see that your `response_ok` function is handling the different types of possible input in any way.  What's going on there?